src/api/macos/mod.rs

```diff
@@ -56,10 +57,21 @@ impl TrayItemMacOS {
         Ok(())
     }
 
-    pub fn set_icon_template(&mut self, icon: &str) -> Result<(), TIError> {
+    pub fn set_icon_template(&mut self, icon:  IconType) -> Result<(), TIError> {
```

Comment:
Would be nice to not break the current API here.

I think that 
```rust
pub fn set_icon_template<I: Into<IconType>>(&mut self, icon I) -> Result<(), TIError> { ... }
```
could work if `IconType` would implement `From<&str>` and `From<&[u8]>`